src/processor/operator/load.cpp

```diff
@@ -0,0 +1,62 @@
+#include "processor/operator/load.h"
+
+#include "common/exception/io.h"
+
+#ifdef _WIN32
+
+#include "windows.h"
+#define RTLD_NOW 0
+#define RTLD_LOCAL 0
+
+inline void* dlopen(const char* file, int mode) {
+    D_ASSERT(file);
+    auto fpath = WindowsUtil::UTF8ToUnicode(file);
+    return (void*)LoadLibraryW(fpath.c_str());
+}
+
+inline void* dlsym(void* handle, const char* name) {
+    D_ASSERT(handle);
+    return (void*)GetProcAddress((HINSTANCE)handle, name);
+}
+
+#else
+#include <dlfcn.h>
+#endif
+
+#include "common/system_message.h"
+#include "main/database.h"
+
+using namespace kuzu::common;
+
+typedef void (*ext_init_func_t)(kuzu::main::Database&);
+
+namespace kuzu {
+namespace processor {
+
+bool Load::getNextTuplesInternal(ExecutionContext* context) {
+    if (hasExecuted) {
+        return false;
+    }
+    hasExecuted = true;
+#ifdef _WIN32
+    auto fpath = WindowsUtil::UTF8ToUnicode(path);
+    auto libHdl = LoadLibraryW(fpath.c_str());
+#else
+    auto libHdl = dlopen(path.c_str(), RTLD_NOW | RTLD_LOCAL);
+#endif
+    if (libHdl == nullptr) {
```

Comment:
Can we add tests for these exceptions?

Reply:
The error message is system dependent, we should improve our testing framework so that it allows fuzzy match error strings.

Reply:
Okay. Can you open an issue on this? We should improve the testing framework for this.